### PR TITLE
Frontend refactor - Using module manager

### PR DESCRIPTION
**Pull request by baiqiushi**
## Overview ##
* `query-util.js`: provides common reusable JSON generating code as previous `common/services.js` did
* `map/controllers.js`: listen to `Leaflet`'s `zoomend` and `dragend` events and do some common geojson files loading work and then `publish` **CHANGE_ZOOM_LEVEL** and **CHANGE_REGION_BY_DRAG** events to `moduleManager`
* `map/countmap.js, heatmap.js, pinmap.js`: all subscribe to `moduleManager`'s **CHANGE_ZOOM_LEVEL**, **CHANGE_REGION_BY_DRAG**, **CHANGE_MAP_TYPE**,  **CHANGE_TIME_SERIES_RANGE** and **CHANGE_SEARCH_KEYWORD** events and send queries through `cloudberryClient` to backend accordingly
* `searchbar/controllers.js`: publish **CHANGE_SEARCH_KEYWORD** event to `moduleManager` according to user actions
* `sidebar/controllers.js`:  subscribe to `moduleManager`'s **CHANGE_ZOOM_LEVEL**, **CHANGE_REGION_BY_DRAG**, **CHANGE_TIME_SERIES_RANGE** and **CHANGE_SEARCH_KEYWORD** events and send queries through `cloudberryClient` to backend accordingly; also publish **CHANGE_MAP_TYPE** event to `moduleManager` according to user actions
* `timeseries/controllers.js`: publish **CHANGE_TIME_SERIES_RANGE** event to `moduleManager` according to user actions

### CODE REVIEW

File: examples/twittermap/web/public/javascripts/map/controllers.js [hunk 1]
```diff
@@ -250,181 +251,79 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
...
-                    $scope.resetGeoIds($scope.bounds, data, 'cityID');
-                    cloudberry.parameters.geoLevel = 'city';
-                    cloudberry.query(cloudberry.parameters);
-                }
+            //set center and boundary done by Cache
+            if (!$scope.status.init) {
+              $scope.resetGeoIds($scope.bounds, data, 'cityID');
+              cloudberry.parameters.geoLevel = 'city';
+              // Publish zoom event to moduleManager
+              moduleManager.publishEvent(moduleManager.EVENT.CHANGE_ZOOM_LEVEL, {level: $scope.map.getZoom(), bounds: $scope.map.getBounds()});
```
[the remark this answers is not in the document]
Comment (baiqiushi) [reply examples/twittermap/web/public/javascripts/map/controllers.js]: Solved

File: examples/twittermap/web/public/javascripts/map/countmap.js [hunk 2]
```diff
@@ -434,18 +510,21 @@ angular.module('cloudberry.map')
     // map type change handler
     // initialize the map (styles, zoom/drag handler, etc) when switch to this map
     // clear the map when switch to other map
-    $rootScope.$on('maptypeChange', function (event, data) {
-      if (cloudberry.parameters.maptype == 'countmap') {
+    function onMaptypeChange(event) {
```
[the remark this answers is not in the document]
Comment (baiqiushi) [reply examples/twittermap/web/public/javascripts/map/countmap.js]: Done.

File: examples/twittermap/web/public/javascripts/map/countmap.js [hunk 3]
```diff
@@ -434,18 +510,21 @@ angular.module('cloudberry.map')
...
+      if (event.cur === "countmap") {
```
[the remark this answers is not in the document]
Comment (baiqiushi) [reply examples/twittermap/web/public/javascripts/map/countmap.js]: Done

File: examples/twittermap/web/public/javascripts/common/query-util.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
+angular.module("cloudberry.common")
```
[the remark this answers is not in the document]
Comment (baiqiushi) [reply examples/twittermap/web/public/javascripts/common/query-util.js]: Different from module manager, I don't what to put here besides these words.

[the remark this answers is not in the document]
Comment (waans11) [reply examples/twittermap/web/public/javascripts/common/query-util.js]: I put these comments before lines 1-5 were added. OK. Done.

File: examples/twittermap/web/public/javascripts/common/query-util.js [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+  .service("queryUtil", function(cloudberryConfig) {
+
+    var queryUtil = {
+
+      startDate: config.startDate,
+      endDate: config.endDate,
+      defaultNonSamplingDayRange: 1500,
+      defaultSamplingDayRange: 1,
+      defaultSamplingSize: 10,
```
[the remark this answers is not in the document]
Comment (waans11) [reply examples/twittermap/web/public/javascripts/common/query-util.js]: OK

**Pull request merged**